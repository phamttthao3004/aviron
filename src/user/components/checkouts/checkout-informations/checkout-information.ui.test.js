import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CONSTANTS from '../../../scripts/apps/constants';
import CheckoutInformationUI from './checkout-information.ui';

describe('<CheckoutInformationUI />', () => {
    it('h1 contains correct text', () => {
        const props = { clickElement: jest.fn() };
        const wrapper = shallow(<CheckoutInformationUI {...props} />);
        expect(wrapper.find('.title').text()).toBe('Shipping information');
    });
    it('Render Checkout UI component matches the snapshot', () => {
        const props = { clickElement: jest.fn() };
        const tree = shallow(<CheckoutInformationUI {...props} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it('Button click go to next step', () => {
        const clickElement = jest.fn();
        const wrapper = shallow(<CheckoutInformationUI clickElement={clickElement} />);
        const history = { push: jest.fn() };
        const pathname = { pathname: CONSTANTS.NAVIGATION_URL.CHECKOUT_PAYMENT_METHOD };
        wrapper.find('.btn-pay').simulate('click');
        expect(clickElement).toHaveBeenCalled(history.push(pathname));
    });
});
