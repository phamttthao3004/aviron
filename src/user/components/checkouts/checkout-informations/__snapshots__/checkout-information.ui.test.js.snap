// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutInformationUI /> Render Checkout UI component matches the snapshot 1`] = `
<div
  className="checkout-content"
>
  <div
    className="title"
  >
    Shipping information
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6 col-xs-12"
    >
      <p>
        First Name
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
    <div
      className="col-sm-6 col-xs-12"
    >
      <p>
        Last Name
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
  </div>
  <p>
    Phone
  </p>
  <div
    className="input-group"
  >
    <span
      className="input-group-addon"
      id="basic-addon2"
    >
      +1
    </span>
    <input
      aria-describedby="basic-addon2"
      className="form-control"
      placeholder=""
      type="text"
    />
  </div>
  <p>
    Address
  </p>
  <div
    className="input-group"
  >
    <input
      aria-describedby="basic-addon2"
      className="form-control"
      placeholder=""
      type="text"
    />
  </div>
  <p>
    Apt, Suite
  </p>
  <div
    className="input-group"
  >
    <input
      aria-describedby="basic-addon2"
      className="form-control"
      placeholder=""
      type="text"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-xs-12"
    >
      <p>
        City
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
    <div
      className="col-sm-4 col-xs-12"
    >
      <p>
        State/Province
      </p>
      <div
        className="input-group"
      >
        <input
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
    <div
      className="col-sm-4 col-xs-12"
    >
      <p>
        Zip/Postal code
      </p>
      <div
        className="input-group"
      >
        <input
          aria-describedby="basic-addon2"
          className="form-control"
          placeholder=""
          type="text"
        />
      </div>
    </div>
  </div>
  <p>
    Country
  </p>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      placeholder=""
      type="text"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <button
        className="btn btn-buy-more"
        type="button"
      >
        Buy more
      </button>
    </div>
    <div
      className="col-xs-6 text-right"
    >
      <button
        className="btn btn-pay"
        onClick={[MockFunction]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
